src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Minigame.kt

```diff
@@ -0,0 +1,122 @@
+
+import com.fasterxml.jackson.core.type.TypeReference
+import io.github.yuutoproject.yuutobot.commands.base.AbstractCommand
+import io.github.yuutoproject.yuutobot.commands.base.CommandCategory
+import io.github.yuutoproject.yuutobot.commands.minigame.MinigameInstance
+import io.github.yuutoproject.yuutobot.commands.minigame.MinigameListener
+import io.github.yuutoproject.yuutobot.objects.Question
+import io.github.yuutoproject.yuutobot.utils.jackson
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionAddEvent
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionRemoveEvent
+
+class Minigame : AbstractCommand(
+    "minigame",
+    CommandCategory.GAME,
+    "Play a fun quiz with your friends!",
+    "Run `minigame` to begin a new game, and react within the countdown to join.\nRun `minigame skip` to skip a question you do not wish to answer."
+) {
+    // String is the channel ID
+    private var minigames = mutableMapOf<Long, MinigameInstance>()
+    private val listener = MinigameListener(this)
+    private val questions: List<Question>
+
+    init {
+        val json = jackson.readTree(this.javaClass.getResource("/minigame.json"))
+        questions = jackson.readValue(json.traverse(), object : TypeReference<List<Question>>() {})
+    }
+
+    override fun run(args: MutableList<String>, event: GuildMessageReceivedEvent) {
+        val id = event.channel.idLong
+
+        // Handling for when a game is already in progress
+        val minigame = minigames[id]
+        if (minigame != null) {
+            // If a user attempts to skip a question
+            if (args.getOrNull(0) == "skip") {
+                if (!minigame.players.contains(event.author.idLong)) {
+                    event.channel.sendMessage("You can't skip a question if you aren't in the game.").queue()
+                    return
+                }
+
+                if (!minigame.begun) {
+                    event.channel.sendMessage("The game has not started yet!").queue()
+                    return
+                }
+
+                minigame.progress(event)
+                return
+            }
+
+            // If the user attempts to start a new game while a game is already in progress,
+            // Either cancel it if it's stale or indicate that a game is already in progress
+            if (System.currentTimeMillis() - minigame.timer > 30_000) {
+                event.channel.sendMessage("Cancelling stale game...").queue()
+                unregister(minigame)
+            } else {
+                event.channel.sendMessage("A game is already running!").queue()
+                return
+            }
+        }
+
+        // Register our listener if it doesn't already exist
+        if (!event.jda.registeredListeners.contains(listener)) {
+            event.jda.addEventListener(listener)
+        }
```

Comment:
Don't do this, register the listener on the builder and inject it into the constructor for this command

Reply:
I'm unsure how to do this without it being very complicated. The listener needs a reference to Minigame in order to call its functions, and Minigame needs to have the Listener as well. How exactly would I go about doing this?

Reply:
Re-write it, I don't see why the minigame command needs access to the listener

Reply:
But this is not acceptable

Reply:
Even if I get rid of the listener from minigame, I still don't know where I'd register the listener. In Yuuto.kt? How would I create the Minigame command itself and give it to the CommandManager? 'This is not acceptable' is not constructive.

Reply:
My suggestion is to drop auto-loading of commands, that way we can more easily inject parameters in constructors